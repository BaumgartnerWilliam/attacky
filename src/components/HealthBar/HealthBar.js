import React from 'react';
import {ProgressBar, Row} from 'react-bootstrap';

const HealthBar = ({ value, isDanger }) => (
  <Row className="justify-content-center">
    <ProgressBar
      data-testid="health-bar"
      data-value={value}
      striped
      animated
      className="w-50"
      variant={(isDanger && 'danger') || 'success'}
      now={value}
    />
  </Row>
);

export default HealthBar;
