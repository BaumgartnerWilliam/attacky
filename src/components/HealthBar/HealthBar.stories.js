import React from 'react';
import { withKnobs, number, boolean } from '@storybook/addon-knobs';

import HealthBar from './HealthBar';

export default {
  title: 'HealthBar',
  decorators: [withKnobs]
};

export const example = () => (
  <HealthBar
    value={number('value', 70)}
    isDanger={boolean('isDanger', false)}
  />
);
