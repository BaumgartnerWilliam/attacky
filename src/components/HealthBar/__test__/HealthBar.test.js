import React from 'react';
import { render } from '@testing-library/react';
import HealthBar from '../HealthBar';

describe('health bar component', () => {
  it('renders without crashing', () => {
    const { container } = render(<HealthBar />);
    expect(container).toBeInTheDocument();
  });

  it('renders the value correctly', () => {
    const { getByTestId } = render(<HealthBar value="5" />);
    const el = getByTestId('health-bar');
    expect(el).toHaveAttribute('data-value', '5');
  });

  it('renders the danger version css', () => {
    const { getByTestId } = render(<HealthBar isDanger={true} />);
    const el = getByTestId('health-bar');
    expect(el.firstChild).toHaveClass('bg-danger');
  });

  it('matches snapshot when danger mode', () => {
    const { getByTestId } = render(<HealthBar value={'5'} isDanger={true} />);
    const testid = getByTestId('health-bar');
    expect(testid).toMatchSnapshot();
  });

  it('matches snapshot when not in danger mode', () => {
    const { getByTestId } = render(<HealthBar value={'99'} />);
    const testid = getByTestId('health-bar');
    expect(testid).toMatchSnapshot();
  });
});
