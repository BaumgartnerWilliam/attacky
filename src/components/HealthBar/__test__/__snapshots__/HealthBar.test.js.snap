// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`health bar component matches snapshot when danger mode 1`] = `
<div
  class="w-50 progress"
  data-testid="health-bar"
  data-value="5"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="5"
    class="progress-bar bg-danger progress-bar-animated progress-bar-striped"
    role="progressbar"
    style="width: 5%;"
  />
</div>
`;

exports[`health bar component matches snapshot when not in danger mode 1`] = `
<div
  class="w-50 progress"
  data-testid="health-bar"
  data-value="99"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="99"
    class="progress-bar bg-success progress-bar-animated progress-bar-striped"
    role="progressbar"
    style="width: 99%;"
  />
</div>
`;
